modin/core/execution/unidist/implementations/pandas_on_unidist/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,566 @@
+
+        Returns
+        -------
+        int
+            The length of the partition.
+        """
+        if self._length_cache is None:
+            if self.axis == 0:
+                self._length_cache = sum(
+                    obj.length() for obj in self.list_of_block_partitions
+                )
+            else:
+                self._length_cache = self.list_of_block_partitions[0].length()
+        return self._length_cache
+
+    _width_cache = None
+
+    def width(self):
+        """
+        Get the width of this partition.
+
+        Returns
+        -------
+        int
+            The width of the partition.
+        """
+        if self._width_cache is None:
+            if self.axis == 1:
+                self._width_cache = sum(
+                    obj.width() for obj in self.list_of_block_partitions
+                )
+            else:
+                self._width_cache = self.list_of_block_partitions[0].width()
+        return self._width_cache
+
+    def drain_call_queue(self, num_splits=None):
+        """
+        Execute all operations stored in this partition's call queue.
+
+        Parameters
+        ----------
+        num_splits : int, default: None
+            The number of times to split the result object.
+        """
+        if len(self.call_queue) == 0:
+            # this implicitly calls `drain_call_queue`
+            _ = self.list_of_blocks
+            return
+        drained = super(PandasOnUnidistDataframeVirtualPartition, self).apply(
+            _DRAIN, num_splits=num_splits, call_queue=self.call_queue
+        )
+        self._list_of_block_partitions = drained
+        self.call_queue = []
+
+    def wait(self):
+        """Wait completing computations on the object wrapped by the partition."""
+        self.drain_call_queue()
+        futures = self.list_of_blocks
+        wait(futures)
+
+    def add_to_apply_calls(self, func, *args, length=None, width=None, **kwargs):
+        """
+        Add a function to the call queue.
+
+        Parameters
+        ----------
+        func : callable or unidist.ObjectRef
+            Function to be added to the call queue.
+        *args : iterable
+            Additional positional arguments to be passed in `func`.
+        length : unidist.ObjectRef or int, optional
+            Length, or reference to it, of wrapped ``pandas.DataFrame``.
+        width : unidist.ObjectRef or int, optional
+            Width, or reference to it, of wrapped ``pandas.DataFrame``.
+        **kwargs : dict
+            Additional keyword arguments to be passed in `func`.
+
+        Returns
+        -------
+        PandasOnUnidistDataframeVirtualPartition
+            A new ``PandasOnUnidistDataframeVirtualPartition`` object.
+
+        Notes
+        -----
+        It does not matter if `func` is callable or an ``unidist.ObjectRef``. Unidist will
+        handle it correctly either way. The keyword arguments are sent as a dictionary.
```

Comment:
this note feels unnecessary IMHO

Reply:
but it brings no real value, stating that "engine works" (this basically is what this note is saying) - one should read unidist docs if feeling unsure, otherwise we'd have to comment about some behaviour all over the place

Reply:
removed